### PR TITLE
Implement metrics for thread pool suppliers (3.0)

### LINKED ISSUE #3639: Metrics for thread pools (3.x)
3.x tracker for #947

### PR DESCRIPTION
**Pull request by tjquinno**
Resolves #3639 

For description see PR #3630 (2.x)

### CODE REVIEW

File: microprofile/lra/jax-rs/src/main/java/io/helidon/microprofile/lra/NonJaxRsResource.java [hunk 1]
```diff
@@ -202,7 +200,8 @@ private void sendResponse(ServerResponse res, Response response) {
         }
     }
 
-    void terminate(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event) {
+   @PreDestroy
+    void terminate() {
```
Comment (tomas-langer) [inline microprofile/lra/jax-rs/src/main/java/io/helidon/microprofile/lra/NonJaxRsResource.java:204]: Why is this changed?

Comment (tjquinno) [reply microprofile/lra/jax-rs/src/main/java/io/helidon/microprofile/lra/NonJaxRsResource.java:204]: In my local tests, I was seeing the class instantiated twice. From the stack traces in the debugger is seemed that CDI was instantiating it the second time in order to notify it of the before-destroyed event. Because the logic in the method was simply cleaning up resources allocated by the bean itself (and not really taking action related to the shut-down of other beans), using `@PreDestroy` not only resolved the problem I was seeing but also more closely aligned with the work the method was doing.

Comment (tjquinno) [reply microprofile/lra/jax-rs/src/main/java/io/helidon/microprofile/lra/NonJaxRsResource.java:204]: @tomas-langer Are you OK with this? Was this your only concern? I would like to merge this PR.

File: metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java [hunk 2]
```diff
@@ -153,4 +154,51 @@ void checkKPIDisabledByDefault() {
...
+                .accept(MediaType.TEXT_PLAIN)
+                .path("greet/slow")
+                .submit()
+                .await();
+
+        assertThat("Slow greet access response status", slowGreetResponse.status().code(), is(200));
+
+        WebClientResponse secondMetricsResponse = metricsRequestBuilder
+                .submit()
+                .await();
```
Comment (danielkec) [inline metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java:194]: There is some internal timeout in the webclient but it is always better to use `await` with some duration

Comment (tjquinno) [reply metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java:194]: Fixed.

File: common/configurable/src/test/java/io/helidon/common/configurable/ScheduledThreadPoolSupplierTest.java [hunk 3]
```diff
@@ -38,20 +41,28 @@
...
+    private static ScheduledThreadPoolSupplier configuredSupplier;
+    private static ScheduledThreadPoolSupplier defaultSupplier;
+
+
     @BeforeAll
     static void initClass() {
-        defaultInstance = ScheduledThreadPoolSupplier.create().getThreadPool();
+        ObserverForTesting.clear();
+        defaultSupplier = ScheduledThreadPoolSupplier.create();
+        defaultInstance = defaultSupplier.getThreadPool();
```
Comment (danielkec) [inline common/configurable/src/test/java/io/helidon/common/configurable/ScheduledThreadPoolSupplierTest.java:53]: This test doesn't seem to be shutting down the executors

**Review by danielkec — CHANGES_REQUESTED** [review]

File: common/configurable/src/test/java/io/helidon/common/configurable/ScheduledThreadPoolSupplierTest.java (hunk 3 above)
Comment (tjquinno) [reply common/configurable/src/test/java/io/helidon/common/configurable/ScheduledThreadPoolSupplierTest.java:54]: Fixed.

**Review by danielkec — APPROVED** [review]
